### ISSUE TITLE
HiDPI Issue

### ISSUE THREAD

**Issue by Imerion** [ONLYOFFICE/DesktopEditors#217]
**Do you want to request a *feature* or report a *bug*?**
Bug report.

**What is the current behavior?**
UI in latest version is way to large (zoomed in) despite not displayed on a HiDPI screen. My screen is 1920x1280, and the software assumes this is a HiDPI mode when it isn't.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
Just start the software on a screen with a 1920x1280 screen.

**What is the expected behavior?**
Software is displayed at normal resolution on a screen with a 1920x1280 resolution.

**Which versions of DesktopEditors, and which OS are affected by this issue? Did this work in previous versions of DesktopEditors?**
Doesn't work in 5.2.8. Worked in version 5.1.27 using Linux Xubuntu 18.10.

Thanks for making this software and releasing it as open source!

**Comment by ShockwaveNN** [comment]
@Imerion Hi, I checked this resolution on my VirtualBox and everything shown fine to me, could you specify your Display model (or notebook model)

**Comment by Imerion** [comment]
@ShockwaveNN Hi! I'm using a Pavilion X2 12 (https://support.hp.com/au-en/document/c05140300). This has happened with a few other programs as well. I have always assumed it was because of my 3:2 monitor. For other apps a system wide setting to disable HiDPI mode for certain toolkits, such as QT, has helped.

Here, I get twice the size I should, making it almost impossible to use the program. If it helps I can attach screenshots.

**Comment by ShockwaveNN** [comment]
@Imerion The reason is that your monitor size is 12'', so our algorithm is detect this as HiDPI screen. We'll try to fix this, but in meantime you can run `onlyoffice-desktopeditors` with `--force-scale=1`

**Comment by Imerion** [comment]
Ah, I see. I guess my monitor falls on the wrong size of that fence. :) But --force-scale=1 works great! Thanks for taking your time!

**Comment by 0x264f** [comment]
I also have a problem with scaling onlyoffice 5.2.8 on Kubuntu 18.04. I use a custom DPI (115) and scaling the system interface 1.2. The display of my laptop is 14", the screen resolution is 1920×1080. 
I tried deb package, snap and appimage, but every time the program was displayed incorrectly: 
<details>
  <summary>Normal launch</summary>
  <img alt="Normal launch" src="https://user-images.githubusercontent.com/34649010/55682650-e49f4b80-593e-11e9-8660-0fabc9111569.png" />
  <img alt="Normal launch" src="https://user-images.githubusercontent.com/34649010/55682658-fc76cf80-593e-11e9-9373-1c6da5aa261e.png" />
</details> 

I also ran the program using the --force-scale flag, but it has fixed values ​​(1,2, ..., n), which was either too small or too large.

I found a solution using [run_scaled](https://github.com/kaueraal/run_scaled), the program interface had correct scaling (using the scaling value 1.4), but due to the specifics of the utility, the interface was blurred: 
<details>
  <summary>Click to expand</summary>
  <img alt="Power Browser NTP" src="https://user-images.githubusercontent.com/34649010/55682714-df8ecc00-593f-11e9-8132-56887a11bc8a.png"/>
</details> 

As a result, I still do not have the perfect fix for this bug, the program remains unusable at the moment. If I could run a program with the --force-scale flag and a value of 1.4, everything would be perfect.

**Comment by articlecat** [comment]
The problem persists on version 5.3.3 on Linux. Using [run_scaled](https://github.com/kaueraal/run_scaled), the following command works on a desktop environment with 200% scaling:

`run_scaled --scale=1 onlyoffice-desktopeditors --force-scale=2`

Although the editing interface scales properly, this workaround has side effects: the mouse pointer (when hovering over OnlyOffice), the file picker, and certain other UI elements are shrunk to a quarter of their normal size.

**Comment by grobbie** [comment]
If you're on ChromeOS, use this: it works for nearly everything except a few QT widgets in the file browser dialog. Looks crisp, no blurry rendering. You need to change the "Exec" line in the file /usr/share/applications/desktop-editors.desktop:
```Exec=env QT_SCREEN_SCALE_FACTORS=1 QT_SCALE_FACTOR=1 QT_FONT_DPI=263 PLASMA_USE_QT_SCALING=1 /usr/bin/sommelier -X --scale=1 --dpi=120 /usr/bin/desktopeditors --force-scale=2 %F```

**Comment by articlecat** [comment]
The run_scaled workaround is no longer working.

**Comment by GaboZero** [comment]
In my case I have a Dell Latitude 9470, it has a 14" screen with a 1080p resolution, when I use the software the tittle bar is too big, and the window does not fit on the screen cutting a part of the program window.

**Comment by archont00** [comment]
Similar here. Ubuntu 19.10, FullHD Samsung TV as display via HDMI on integrated Intel J4105-ITX.

X.org correctly set Full HD resolution and dpi 96x96.

OnlyOffice destkop does not fit the screen though, it is scaled by factor of two. Running with --force-scale=1 helps, but then it it is a bit too small.

Another problem is that OnlyOffice is a snap package, so I still have to learn how to modify/create .desktop file to have the app started with --force-scale=1 for every user on the computer.

**Comment by BjoernDaase** [comment]
Unfortunately also can confirm this in F31. Happens if installed via flatpak, snap or dnf...

**Comment by GimmeLewd** [comment]
--force-scale=1 work for me, but is it possible to fix this issue?
Because sometimes it's giving me big scale title bar.

**Comment by PapsOu** [comment]
I've got the same wrong behavior under fedora-31
![Peek 05-12-2019 17-29](https://user-images.githubusercontent.com/5792207/70254158-ea2a8b00-1784-11ea-85ed-d075d92289c7.gif)

**Comment by rghv** [comment]
> Unfortunately also can confirm this in F31. Happens if installed via flatpak, snap or dnf...
> 
> At least, the `--force-scale=1` workaround works for me.

Can confirm. `--force-scale=1` makes it usable, though the experience is not all too great, it's much bearable as opposed to without it. 

Tested on below listed machine, OS specs.

```
System:    Host: XXXXXXXX Kernel: 4.20.17-042017-generic x86_64 bits: 64 compiler: gcc v: 8.3.0 Desktop: Xfce 4.12.3 

Distro: Linux Mint 19.2 Tina base: Ubuntu 18.04 bionic 

Machine:   Type: Laptop System: Dell product: Latitude 5280 v: N/A serial: <filter> 

Mobo: Dell model: N/A serial: <filter> UEFI: Dell v: 1.16.0 date: 07/03/2019 

CPU:       Topology: Dual Core model: Intel Core i5-7300U bits: 64 type: MT MCP arch: Kaby Lake rev: 9 L2 cache: 3072 KiB flags: lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx bogomips: 21696  Speed: 2699 MHz min/max: 400/3500 MHz Core speeds (MHz): 1: 2400 2: 2468 3: 2429 4: 2461 

Graphics:  Device-1: Intel HD Graphics 620 vendor: Dell driver: i915 v: kernel bus ID: 00:02.0 Display: x11 server: X.Org 1.19.6 driver: modesetting unloaded: fbdev,vesa resolution: 1920x1080~60Hz OpenGL: renderer: Mesa DRI Intel HD Graphics 620 (Kaby Lake GT2) v: 4.5 Mesa 19.0.8 direct render: Yes
```

**Comment by articlecat** [comment]
The issue persists on version 5.5.1 on Linux. When maximized, the interface looks like the screenshots posted in issue #136:

![](https://user-images.githubusercontent.com/171169/61226053-5f6ac600-a6e7-11e9-8c15-bc9ab55b36dc.png)

![](https://user-images.githubusercontent.com/8462696/61990480-647f2c00-b07c-11e9-9094-1df15fb52c1f.png)

![](https://user-images.githubusercontent.com/8481428/66786259-1130b300-eee0-11e9-9f0a-237b9e5aea38.png)

**Comment by timrichardson** [comment]
This is happening to me. I have three monitors on my linux PC. One of them is 4K but it is not hidpi. It is just a very big screen. On this screen, OnlyOffice is scaling as it it was a HiDPI screen.

**Comment by mathewng** [comment]
I have a 4k display and it worked pretty well with
QT_SCREEN_SCALE_FACTORS=0.5 QT_SCALE_FACTOR=2 DesktopEditors-x86_64.AppImage --force-scale=2

Everything is doubled in size which is what I want moving up from 1080p to 4k

**Comment by JOduMonT** [comment]
@BjoernDaase thanks for the tips
> At least, the `--force-scale=1` workaround works for me too.

#### Desktop
OS: Debian Bullseye
KDE Plasma: 5.17.5
Graphic card driver: i915
Display recognized as : LG TV
resolution: 1920x1080

This is also an issue with my laptop when it is connected to the same screen

#### Laptop
OS: Debian Bullseye (but I also tried with an xubuntu livecd 18.04LTS and 20.04LTS)
XFCE 4.14
Graphic card driver: i915
Display recognized as : LG TV
resolution: 1920x1080

**Comment by despens** [comment]
Using the snap version 5.5.1 on Ubuntu 20.04, on a 2560×1440 pixel 14", and same resolution 27" display. In both cases, the interface appears scaled to 200%, and in both cases, the display scaling of the OS is set to 100%. It would be helpful to present an option to users for setting the scaling themselves, or to just use the Scale value of the desktop environment.

**Comment by asbl** [comment]
Same here: I'm using Manjaro on Thinkpad yoga 3rd ed. With ubuntu i can Onlyoffice without problems. In Manjaro i can only use Onlyoffice without scaling-display-error, if I reduce my Resolution to 1600x900.

**Comment by IGBC** [comment]
I am getting a 2x HiDPI factor applied on my 14" 1440p monitor

I don't consider this a bug, 1440p is widely considered HiDPI.

What I do consider a bug is that there is no GUI option to override this behaviour, as HiDPI makes an (incorrect) assumption about how I use my computer. 

I wish to weigh in and request a proper setting be added, as launching the program from the command line is not a sustainable solution for an office program.

**Comment by cgku** [comment]
I too would like to see this adjustable within the application. 

My problem is exactly that as described by @articlecat; 4k screen, 2x scaling and only partial screen rendering.
Using the mitigation by @mathewng I am able to update the .desktop file and use the program.  Unfortunately, dialog windows are way out of scale.
![image](https://user-images.githubusercontent.com/54686681/93688156-a9cc9d00-fa78-11ea-8d5a-414edf6790d1.png)
These windows are not as severely distorted in other applications, so I assume this is a side affect from the workaround doubling the scale of what the DE was already scaling.

My system:
- Operating System: Kubuntu 20.04
- KDE Plasma Version: 5.18.5
- KDE Frameworks Version: 5.68.0
- Qt Version: 5.12.8
- Kernel Version: 5.4.0-47-generic
- Display:  3840x2160 (using 200% scaling in KDE)
- Using AppImage, same problem found with Snap

While I would agree that ultimately the app should let the DE do it's job when possible, my experience is that only apps written in the same environment as the desktop (ie: GTK+ or QT) do a decent job of this.  Too many apps don't honor scaling from the DE and are either unusable or difficult to use on HiDPI screens.  Using workarounds at launch can be too brute-force, leading to other problems as seen with this dialog box issue, and require even more work to update all the various contextual menus that would be missing these switches in their statements.  Given the mix-and-match nature of the Linux world it only makes sense for the application to not make too many assumptions.  The programs I've used with the best management of this problem have included an option to override settings in-app, with the default set to inherit the DE settings.

Thanks for your consideration.

**Comment by Chrysostomus** [comment]
Having an option to adjust the scaling or even to just plain disable would be a blessing. The --force-scale=1 flag no longer works. I have 1920x1080 screen, and onlyoffice detects it as hidpi. The scaled view makes it practically unusable. I can trick it into running normally by setting resolution to 1280x720, starting onlyoffice and restoring the normal resolution. But that is a bit too much trouble for simply starting a text editor.

**Comment by NicTanghe** [comment]
I have no idea but it is completly horible that i cant use only office becouse the ui scales To extremely zoomed in.

**Comment by texretrotech** [comment]
The following command resolved the issue for me:
QT_FONT_DPI=163 QT_SCREEN_SCALE_FACTORS=1 QT_SCALE_FACTOR=1 onlyoffice-desktopeditors --force-scale=2

**Comment by cgku** [comment]
This issue seems to be resolved in OnlyOffice 6.2.0

**Comment by ShockwaveNN** [comment]
@cgku I think I agree with you

This issue is collection of different, somewhat directly unrelated issues about interface scale

We've fixed most of them with introduction of option to scale interface

I'm sure that not all issues are fixed and some of them is still actual, but I think for not fixed issues is better to create new one, because this issue is very hard to parse

So I close this issue but please feel free to comment and create new issues

**Comment by proteriax** [comment]
The option to scale interface only goes up to 200% while a 4K screen needs 300%. Would it be possible to add this option as well?

**Comment by ShockwaveNN** [comment]
@proteriax Hi, could you create a seprate issue for 300%

**Comment by themooleman** [comment]
Following on from this topic (but still related)....

I specify DPI scaling using my .Xresources file as it is easy and broadly respected by applications. Is there a way to disable ONLYOFFICE's automatic scaling and have is respect a .Xresources value e.g. `Xft.dpi: 130`?

For now I've simply copied and modified onlyoffice-desktopeditors.desktop and added `--force-scale=1.2` to the exec line but I work between two different setups and when I'm connected to a 4K 27" display, this causes ONLYOFFICE to be far too small. Not a perfect solution. 

If it could respect an Xresources configuration, this wouldn't be an issue as I have a script which swaps out my Xresources (via symlinking) according to whether or not I'm connected to an external display.

Picky, I know.

**Comment by crojack** [comment]
I've tried everything- and simply can't get OnlyOffice to scale properly. Not even on native screen resolution, which is 2K.
![OnlyOfficeScaling](https://user-images.githubusercontent.com/49529443/176758993-41cd0895-8f9e-45a8-8204-cc997cb1b9a3.png)

**Comment by Batwam** [comment]
similar issue here with onlyoffice-desktopeditors ver. 7.1.1.57 selecting scaling in Auto mode equivalent to 175% on a 1920x1200 res screen. Optimal size should be 100% so `--force-scale=1` works but this should ideally work out of the box. Is there a problem with resolution recognotion which I can help troubleshoot?

**Comment by cipitaua** [comment]
I would need options also below 100%, e.g. 50%, 60%, 70%, 80%, 90%.  Otherwise in my case I have to launch it every time with --force-scale=1

**Comment by melodie11** [comment]
Hi 
I met with this issue using OnlyOffice Desktop Editors is 9.1.0.173.

My laptop is a Thinkpad T15 (gen2) with HiDPI running Xubuntu  24.04.

```
$ xrandr 
Screen 0: minimum 320 x 200, current 2880 x 1620, maximum 16384 x 16384
eDP-1 connected primary 2880x1620+0+0 (normal left inverted right x axis y axis) 344mm x 194mm
   3840x2160     60.00 +  60.00
```
(tweaked with Xubuntu display tool to get it to the right font sizes)

Texretrotech command line worked for me `QT_FONT_DPI=163 QT_SCREEN_SCALE_FACTORS=1 QT_SCALE_FACTOR=1 onlyoffice-desktopeditors --force-scale=2`.
